python_modules/libraries/dagster-k8s/dagster_k8s/pipes.py

```diff
@@ -80,24 +83,94 @@ def read_messages(
         finally:
             self._handler = None
 
+    def _get_consume_logs_request_timeout(self) -> Optional[int]:
+        request_timeout_env_var = os.getenv("DAGSTER_K8S_PIPES_CONSUME_POD_LOGS_REQUEST_TIMEOUT")
+        if request_timeout_env_var:
+            return int(request_timeout_env_var)
+
+        # Timeout and reconnect every 10 minutes by default
+        return 600
+
     def consume_pod_logs(
         self,
+        context: Union[OpExecutionContext, AssetExecutionContext],
         core_api: kubernetes.client.CoreV1Api,
         pod_name: str,
         namespace: str,
     ):
+        last_seen_timestamp = None
+        catching_up_after_retry = False
+
+        request_timeout = self._get_consume_logs_request_timeout()
+
+        retries_remaining = int(
+            os.getenv(
+                "DAGSTER_K8S_PIPES_CONSUME_POD_LOGS_RETRIES", DEFAULT_CONSUME_POD_LOGS_RETRIES
+            )
+        )
+
         handler = check.not_none(
             self._handler, "can only consume logs within scope of context manager"
         )
-        for line in core_api.read_namespaced_pod_log(
-            pod_name,
-            namespace,
-            follow=True,
-            _preload_content=False,  # avoid JSON processing
-        ).stream():
-            log_chunk = line.decode("utf-8")
-            for log_line in log_chunk.split("\n"):
-                extract_message_or_forward_to_stdout(handler, log_line)
+
+        while True:
+            if catching_up_after_retry:
+                # ensure that we cover the full time period if the request timed out
+                since_seconds = 2 * request_timeout if request_timeout else None
+            else:
+                since_seconds = None
```

Comment:
added this since_seconds logic which should help? now it will do a rolling window of logs instead of redoing a full scan every time